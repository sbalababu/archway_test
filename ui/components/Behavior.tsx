import * as React from 'react';
import { Card, Icon, List } from 'antd';

interface Props {
  behaviorKey: string;
  icon: string;
  title: string;
  description?: string;
  useCases?: string[];
  selected?: boolean;
  onChange: (behavior: string, selected: boolean) => void;
}

const UseCase = (item: string) => <div>{item}</div>;

const Behavior = ({ behaviorKey, icon, title, description, useCases, selected, onChange }: Props) => (
  <Card
    onClick={() => onChange(behaviorKey, !selected)}
    hoverable={true}
    style={{ flex: 1, borderRadius: 3, marginBottom: 15 }}
    bodyStyle={{ textAlign: 'center' }}
  >
    <div
      style={{
        opacity: selected ? 1 : 0,
        position: 'absolute',
        top: 25,
        right: 25,
        transition: 'opacity 300ms',
      }}
    >
      <Icon type="check" style={{ fontSize: 14 }} />
    </div>
    <Icon type={icon} style={{ fontSize: 42 }} />
    <div style={{ fontSize: 14, textTransform: 'uppercase', display: 'flex', justifyContent: 'center' }}>{title}</div>
    {description && <p>{description}</p>}
    {useCases && (
      <div>
        <div style={{ textTransform: 'uppercase', fontSize: 12 }}>example use cases</div>
        <List dataSource={useCases} renderItem={UseCase} />
      </div>
    )}
  </Card>
);

export default Behavior;
