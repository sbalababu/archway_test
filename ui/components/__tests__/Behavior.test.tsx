import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Behavior from '../Behavior';

describe('Behavior', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Behavior behaviorKey="" icon="" title="" onChange={() => null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
