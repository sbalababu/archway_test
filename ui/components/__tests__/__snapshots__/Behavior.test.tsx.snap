// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior renders correctly 1`] = `
<Card
  bodyStyle={
    Object {
      "textAlign": "center",
    }
  }
  hoverable={true}
  onClick={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "flex": 1,
      "marginBottom": 15,
    }
  }
>
  <div
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
        "right": 25,
        "top": 25,
        "transition": "opacity 300ms",
      }
    }
  >
    <Icon
      style={
        Object {
          "fontSize": 14,
        }
      }
      type="check"
    />
  </div>
  <Icon
    style={
      Object {
        "fontSize": 42,
      }
    }
    type=""
  />
  <div
    style={
      Object {
        "display": "flex",
        "fontSize": 14,
        "justifyContent": "center",
        "textTransform": "uppercase",
      }
    }
  />
</Card>
`;
